vita3k/util/include/util/semaphore.h

```diff
@@ -0,0 +1,34 @@
+#include <condition_variable>
+#include <mutex>
+
+namespace sync_utils {
+class Semaphore {
+private:
+    std::mutex mutex_;
+    std::condition_variable condition_;
+    unsigned long count_ = 0; // Initialized as locked.
```

Comment:
Be specific with your size and use `uint64_t` maybe?

Reply:
👍